src/Components/Components/src/Rendering/RenderedHtmlResult.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Components.Rendering
+{
+    /// <summary>
+    /// Represents the result of rendering a component into static html.
+    /// </summary>
+    public struct RenderedHtmlResult
```

Comment:
This will evolve after preview4 into a more "abstract" type that will simply encapsulate writing the component to a provided stream.

Reply:
That will avoid any double buffering and will not expose the list of tokens.